src/openrct2/actions/LandSetHeightAction.hpp

```diff
@@ -134,12 +134,15 @@ DEFINE_GAME_ACTION(LandSetHeightAction, GAME_COMMAND_SET_LAND_HEIGHT, GameAction
                     zCorner += 2;
                 }
             }
-            if (!map_can_construct_with_clear_at(
-                    { _coords, _height * COORDS_Z_STEP, zCorner * COORDS_Z_STEP }, &map_set_land_height_clear_func,
-                    { 0b1111, 0 }, 0, nullptr, CREATE_CROSSING_MODE_NONE))
+
+            auto clearResult = MapCanConstructWithClearAt(
+                { _coords, _height * COORDS_Z_STEP, zCorner * COORDS_Z_STEP }, &map_set_land_height_clear_func, { 0b1111, 0 },
+                0, CREATE_CROSSING_MODE_NONE);
+            if (clearResult->Error != GA_ERROR::OK)
             {
                 return std::make_unique<GameActionResult>(
-                    GA_ERROR::DISALLOWED, STR_NONE, gGameCommandErrorText, gCommonFormatArgs);
+                    GA_ERROR::DISALLOWED, STR_NONE, clearResult->ErrorMessage.GetStringId(),
```

Comment:
Is copying all the data into a new result on purpose?

Reply:
Yes

Reply:
It doesn't copy all of the data it doesn't copy in the cost for example.

Reply:
Right.